test/suite.js

```diff
@@ -72,6 +72,19 @@ class Suite {
     const test = new Test(name, cb)
     this._queue.push(test)
   }
+
+  testAsync (name, action) {
+    const test = new Test(name, cb => {
+      try {
+        Promise.resolve(action())
+          .then(() => cb(null))
+          .catch((err) => cb(err))
```

Comment:
This can call `cb` a second time if it throws. How about:

```js
Promise.resolve()
  .then(action)
  .then(() => { cb(null) }, cb)
```

Reply:
Oh yes this is way better. In fact I don't think the double wrapped try/catch for sync errors is needed either as the `action` will get invoked via regular Promise semantics and auto convert it to a rejection.